### PR TITLE
[HappyPath] Make checking running application more stable

### PR DESCRIPTION
**Pull request by SkorikSergey**
### What does this PR do?
This PR adds checking that application is ready to use before opening it in the **Preview** panel.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/13979

### CODE REVIEW

File: e2e/pageobjects/ide/Ide.ts [hunk 1]
```diff
@@ -229,6 +231,38 @@ export class Ide {
         await this.driverHelper.waitVisibility(selectedRightToolbarButtonLocator, timeout);
     }
 
+    async getApplicationUrl(notificationText: string) {
```
Comment (Ohrimenko1988) [inline e2e/pageobjects/ide/Ide.ts]: On my opinion for avoiding possible confusion better rename to 
"getApplicationUrlFromNotification"

File: e2e/pageobjects/ide/Ide.ts [hunk 2]
```diff
@@ -229,6 +231,38 @@ export class Ide {
...
+         await this.driverHelper.getDriver().wait(async () => {
+            try {
+                let res = await axios.get(url);
+
+                if (res.status === 200) {
+                    console.log('Application is ready for use.');
+                    return true;
+                }
+            } catch (error) {
+               console.log('Application is not yet ready for use');
```
Comment (Ohrimenko1988) [inline e2e/pageobjects/ide/Ide.ts:259]: What about specifying the exact types of errors?

**Comment by Ohrimenko1988** [comment]
CQ for axios 0.17.1 <https://dev.eclipse.org/ipzilla/show_bug.cgi?id=15800>

**Review by Ohrimenko1988 — APPROVED** [review]

**Comment by dmytro-ndp** [comment]
@SkorikSergey : can you, please, take commit https://github.com/eclipse/che/commit/c732524e4d24cd8e760825f96c927a6229e770f7 from master so that Happy path test could be run completely?

**Comment by dmytro-ndp** [comment]
ci-build

**Review by dmytro-ndp — APPROVED** [review]
Looks good.

File: e2e/pageobjects/ide/Ide.ts [hunk 3]
```diff
@@ -229,6 +231,38 @@ export class Ide {
...
+        let regexp: RegExp = new RegExp('^.*(https?://.*)$');
+
+        if (!regexp.test(notification)) {
+            throw new Error('Cannot obtaine url from notification message');
+        }
+
+        return notification.split(regexp)[1];
+    }
+
+    async waitApplicationReadyToUse(url: string,
```
Comment (rhopp) [inline e2e/pageobjects/ide/Ide.ts]: I would add a verb here to the name... For example `waitApllicationIsReady`.

**Comment by SkorikSergey** [comment]
crw-ci-test

**Review by rhopp — APPROVED** [review]

**Pull request merged**